pkg/lockfile/parse-pnpm-lock.go

```diff
@@ -23,7 +23,7 @@ type PnpmLockPackage struct {
 }
 
 type PnpmLockfile struct {
-	Version  float64                    `yaml:"lockfileVersion"`
+	Version  string                     `yaml:"lockfileVersion"`
```

Comment:
Looks like with latest PNPM lock file version, they changed the format of the version from a float to string. The float64 type here cannot be mapped to the v6 lock file. The string type however maps successfully to both lock file formats.

One option would be to define a new type for the v6+ lock file version but that does not seem elegant.

Happy for your suggestions on this one (I am still fairly new to Go so I may not be aware of a nicer solution to handle this).

Reply:
To avoid breaking that public interface, a new local interface could be added, and then inside the `ParsePnpmLock` function, explicitly convert the version to `float64`.

Something like the following maybe: (I have included just the key changes below)

(The following passes the test suite)

```go
type PnpmLockfile struct {
	Version  float64                    `yaml:"lockfileVersion"`
	Packages map[string]PnpmLockPackage `yaml:"packages,omitempty"`
}

type pnpmLockfileV6 struct {
	Version  string                     `yaml:"lockfileVersion"`
	Packages map[string]PnpmLockPackage `yaml:"packages,omitempty"`
}

func ParsePnpmLock(pathToLockfile string) ([]PackageDetails, error) {
	var parsedLockfileV6 *pnpmLockfileV6
	var parsedLockFile *PnpmLockfile

	lockfileContents, err := os.ReadFile(pathToLockfile)

	if err != nil {
		return []PackageDetails{}, fmt.Errorf("could not read %s: %w", pathToLockfile, err)
	}

	err = yaml.Unmarshal(lockfileContents, &parsedLockfileV6)

	if err != nil {
		return []PackageDetails{}, fmt.Errorf("could not parse %s: %w", pathToLockfile, err)
	}

	parsedVersion, err := strconv.ParseFloat(parsedLockfileV6.Version, 64)

	if err != nil {
		return []PackageDetails{}, fmt.Errorf("could not parse version %s: %w", parsedLockfileV6.Version, err)
	}

	parsedLockFile = &PnpmLockfile{
		Version:  parsedVersion,
		Packages: parsedLockfileV6.Packages,
	}

	return parsePnpmLock(*parsedLockFile), nil
}
```

Reply:
Thanks! That looks pretty reasonable to me. @G-Rath any thoughts?

Reply:
I'd probably be tempted to try using `UnmarshalYAML` instead so that way `PnpmLockfile` by itself can "parse" v6 lockfiles - otherwise this only fixes the problem for when you're using `ParsePnpmLock` (which arguably is not useful since that doesn't expose the `PnpmLockfile` in anyway)

Reply:
That's done, I believe I got the interface implemented correctly. Tests seem to be all passing.